### PR TITLE
[chain] Defer State Root Generation

### PR DESCRIPTION
**Pull request by patrick-ogrady**
Resolves: #399 

### TODO
- [x] update readme with explanation of how this works (block contains root of parent post-execution)
  - [x] update `block.StatefulBlock` comment 
- [x] ensure state sync logic isn't broken (accepted state at end is that of parent, not of accepted block)
- [x] add metric for waiting for root
- [x] limit concurrency of `x/merkledb` to prevent thrashing: https://github.com/ava-labs/avalanchego/pull/1936
  - [ ] ~ensure only 1 root is being generated at a time (merkledb may try to generate multiple roots at once, exceeding the CPU limit) -> run queue in `vm` (need to prevent out-of-band calls to `GetMerkleRoot`)~
- [x] add accept depth to load test: https://github.com/ava-labs/hypersdk/pull/424
- [x] Verify signatures are correct before creating child view
- [ ] ~ensure 1 block above sync is added (need parent block to verify a block) or allow for fetching missing blocks in certain circumstances over the wire~ put all data required to verify a block in state (so don't need parent block)
- [x] Remove dependency on parent block during verification: https://github.com/ava-labs/hypersdk/pull/427

### `Block.Build` Latency
_We use `block.Build` for this comparison because all signatures are pre-verified (when a tx is added to the mempool) and we just want to test the savings when root generation is deferred with no other async jobs running (i.e. tx verification)._

#### Before
```
[08-31|13:30:01.937] INFO load/load_test.go:339 block generation {"t": "4.546424333s", "avg(ms)": 96, "tps": 109976.53614748943}
```

#### After (~22.9% reduction)
```
[08-31|13:28:25.782] INFO load/load_test.go:347 block generation {"t": "3.488638958s", "avg(ms)": 74, "tps": 143322.36898674225}
```

### CODE REVIEW

**Comment by patrick-ogrady** [comment]
initial status...not too much faster because wait signatures is much slower (all cores used by hashing): 
image

File: chain/block.go [hunk 1]
```diff
@@ -708,40 +728,95 @@ func (b *StatelessBlock) Processed() bool {
...
+// run verification depending on the value of [blockRoot].
 //
 // Invariant: [View] with [verify] == true should not be called concurrently, otherwise,
 // it will result in undefined behavior.
-func (b *StatelessBlock) View(ctx context.Context, verify bool) (state.View, error) {
+func (b *StatelessBlock) View(ctx context.Context, blockRoot *ids.ID) (state.View, error) {
 	ctx, span := b.vm.Tracer().Start(ctx, "StatelessBlock.View",
 		oteltrace.WithAttributes(
-			attribute.Bool("verify", verify),
+			attribute.Stringer("blockRoot", blockRoot),
```
Comment (StephenButtolph) [inline chain/block.go:742]: Does this work? The stringer for `ids.ID` takes in an `ids.ID` - not an `*ids.ID`... so I feel like `blockRoot = nil` should cause this to panic... no?

File: vm/vm.go [hunk 2]
```diff
@@ -213,8 +224,15 @@ func (vm *VM) Initialize(
...
-	// Setup worker cluster
-	vm.workers = workers.NewParallel(vm.config.GetParallelism(), 100)
+	// Setup worker cluster for verifying signatures
+	//
+	// If [parallelism] is odd, we assign the extra
+	// core to signature verification.
+	sigParallelism := parallelism - rootGenParallelism
+	if sigParallelism <= 0 {
+		sigParallelism = 1
+	}
```
Comment (StephenButtolph) [inline vm/vm.go:234]: Can we use `math.Max` here?

File: vm/vm.go [hunk 3]
```diff
@@ -199,12 +200,22 @@ func (vm *VM) Initialize(
 	}
 
 	// Instantiate DBs
+	parallelism := vm.config.GetParallelism()
+	rootGenParallelism := parallelism / 2
+	if rootGenParallelism == 0 {
+		rootGenParallelism = 1
+	}
```
Comment (StephenButtolph) [inline vm/vm.go:207]: math.Max?

File: chain/block.go (hunk 1 above)
Comment (patrick-ogrady) [reply chain/block.go:742]: I have no idea.

**Comment by patrick-ogrady** [comment]
Tested flow for processing:
```
[sync] [08-27|22:15:18.858] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/vm.go:634 parsed block {"id": "jBXME9wSHeNAzxfAbC5Ew95FgD9ddA8yJaTaVSkjM6ink5Arv", "height": 1169}
[bootstrap] [08-27|22:15:18.858] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/vm.go:634 parsed block {"id": "jBXME9wSHeNAzxfAbC5Ew95FgD9ddA8yJaTaVSkjM6ink5Arv", "height": 1169}
[bootstrap] [08-27|22:15:18.860] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> builder/time.go:86 notifying to build without waiting {"txs": 0}
[bootstrap] [08-27|22:15:18.860] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/resolutions.go:114 verified block {"blkID": "jBXME9wSHeNAzxfAbC5Ew95FgD9ddA8yJaTaVSkjM6ink5Arv", "height": 1169, "txs": 0, "state ready": true, "unit prices": [100,100,100,100,100], "units consumed": [0,0,0,0,0]}
[bootstrap] [08-27|22:15:18.860] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:625 merkle root generated {"height": 1169, "blkID": "jBXME9wSHeNAzxfAbC5Ew95FgD9ddA8yJaTaVSkjM6ink5Arv", "root": "KHVmuC1P8qeJwB5JLATP2s4Wg7i1ofLb8MCve5hp4Htg28wep"}
[sync] [08-27|22:15:18.860] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:769 block not processed {"height": 1168, "blkID": "2tbuHvaUjfqQXGB4NZBtun8Py8pHQADw3vvLPviJMoRene1zYr", "attemptVerify": true}
[sync] [08-27|22:15:18.860] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:806 verifying block when view requested {"height": 1168, "blkID": "2tbuHvaUjfqQXGB4NZBtun8Py8pHQADw3vvLPviJMoRene1zYr"}
[sync] [08-27|22:15:18.860] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:769 block not processed {"height": 1167, "blkID": "R3ZEtNEE284AQjJnHP45DppXyMc2kVw2uwdBTpH4kguBPYcjf", "attemptVerify": true}
[sync] [08-27|22:15:18.860] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> builder/time.go:86 notifying to build without waiting {"txs": 0}
[sync] [08-27|22:15:18.860] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:625 merkle root generated {"height": 1168, "blkID": "2tbuHvaUjfqQXGB4NZBtun8Py8pHQADw3vvLPviJMoRene1zYr", "root": "Uukwy5NMv6rHYHNZrN8bwPjXF1tUHrzJYuv3gr6Xo45ZUynG"}
[sync] [08-27|22:15:18.860] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/resolutions.go:114 verified block {"blkID": "jBXME9wSHeNAzxfAbC5Ew95FgD9ddA8yJaTaVSkjM6ink5Arv", "height": 1169, "txs": 0, "state ready": true, "unit prices": [100,100,100,100,100], "units consumed": [0,0,0,0,0]}
```

**Comment by patrick-ogrady** [comment]
Hitting error case:
```
[sync] [08-27|22:19:43.201] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:769 block not processed {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw", "attemptVerify": true}
[sync] [08-27|22:19:43.201] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:806 verifying block when view requested {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw"}
[sync] [08-27|22:19:43.201] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:428 could not get parent {"id": "Zi7fWRJbasVW5xaX5Xb1m9kqzWSDAPgEWxbiXCrwQfzujGiE2"}
[sync] [08-27|22:19:43.880] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/warp_manager.go:99 checked for ready jobs {"pending": 0}
[bootstrap] [08-27|22:19:43.992] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/warp_manager.go:99 checked for ready jobs {"pending": 0}
[sync] [08-27|22:19:44.412] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:769 block not processed {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw", "attemptVerify": true}
[sync] [08-27|22:19:44.413] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:806 verifying block when view requested {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw"}
[sync] [08-27|22:19:44.413] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:428 could not get parent {"id": "Zi7fWRJbasVW5xaX5Xb1m9kqzWSDAPgEWxbiXCrwQfzujGiE2"}
[sync] [08-27|22:19:44.557] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:769 block not processed {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw", "attemptVerify": true}
[sync] [08-27|22:19:44.557] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:806 verifying block when view requested {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw"}
[sync] [08-27|22:19:44.557] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:428 could not get parent {"id": "Zi7fWRJbasVW5xaX5Xb1m9kqzWSDAPgEWxbiXCrwQfzujGiE2"}
[sync] [08-27|22:19:44.645] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/vm.go:411 node is not ready yet
[sync] [08-27|22:19:44.707] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:769 block not processed {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw", "attemptVerify": true}
[sync] [08-27|22:19:44.707] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:806 verifying block when view requested {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw"}
[sync] [08-27|22:19:44.707] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:428 could not get parent {"id": "Zi7fWRJbasVW5xaX5Xb1m9kqzWSDAPgEWxbiXCrwQfzujGiE2"}
[sync] [08-27|22:19:44.872] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:769 block not processed {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw", "attemptVerify": true}
[sync] [08-27|22:19:44.872] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:806 verifying block when view requested {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw"}
[sync] [08-27|22:19:44.872] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:428 could not get parent {"id": "Zi7fWRJbasVW5xaX5Xb1m9kqzWSDAPgEWxbiXCrwQfzujGiE2"}
[sync] [08-27|22:19:44.879] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/warp_manager.go:99 checked for ready jobs {"pending": 0}
[bootstrap] [08-27|22:19:44.992] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/warp_manager.go:99 checked for ready jobs {"pending": 0}
[sync] [08-27|22:19:45.024] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:769 block not processed {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw", "attemptVerify": true}
[sync] [08-27|22:19:45.024] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:806 verifying block when view requested {"height": 1461, "blkID": "ko5wqYYFbouJVizWt12C9gLEHn7AsE9EHhRGkEdoyhHeiNuTw"}
[sync] [08-27|22:19:45.024] DEBUG <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> chain/block.go:428 could not get parent {"id": "Zi7fWRJbasVW5xaX5Xb1m9kqzWSDAPgEWxbiXCrwQfzujGiE2"}
[sync] [08-27|22:19:45.684] INFO <nLAmPNCErV2ZUjitfvY5HWdkTvqn2KKBM9BgnSmD1kZ4SWxdS Chain> vm/vm.go:634 parsed block {"id": "4uNC5gYwuNTBUZ9Fc23DgdSmXDhRGqpoUwybFvga4FJVZth1q", "height": 1511}
```

File: vm/vm.go [hunk 4]
```diff
@@ -213,8 +222,12 @@ func (vm *VM) Initialize(
 		return err
 	}
 
-	// Setup worker cluster
-	vm.workers = workers.NewParallel(vm.config.GetParallelism(), 100)
+	// Setup worker cluster for verifying signatures
+	//
+	// If [parallelism] is odd, we assign the extra
+	// core to signature verification.
+	sigParallelism := parallelism - rootGenParallelism
```
Comment (StephenButtolph) [inline vm/vm.go:229]: `rootGenParallelism` can be 0 here... is that expected?

Comment (patrick-ogrady) [reply vm/vm.go:229]: no

Comment (patrick-ogrady) [reply vm/vm.go:229]: have a pending commit that cleans this up

**Comment by patrick-ogrady** [comment]
Need to see at last 1 block past sync target to be able to handle verify:
```
[sync] [08-27|22:34:04.966] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/syncervm_server.go:47 parsed state summary {"summary": "1168:2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT root=2gKBrSishaN73WRRybDV2TBU1Fkk1xK5DemCRs6GnUxo7J2DsF"}
[sync] [08-27|22:34:04.966] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/auth_batch.go:66 enqueued batch for processing during done
[sync] [08-27|22:34:04.966] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/vm.go:634 parsed block {"id": "2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT", "height": 1168}
[sync] [08-27|22:34:04.966] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/auth_batch.go:66 enqueued batch for processing during done
[sync] [08-27|22:34:04.966] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/syncervm_server.go:47 parsed state summary {"summary": "1168:2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT root=2gKBrSishaN73WRRybDV2TBU1Fkk1xK5DemCRs6GnUxo7J2DsF"}
[sync] [08-27|22:34:04.966] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/auth_batch.go:66 enqueued batch for processing during done
[sync] [08-27|22:34:04.967] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/syncervm_server.go:47 parsed state summary {"summary": "1168:2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT root=2gKBrSishaN73WRRybDV2TBU1Fkk1xK5DemCRs6GnUxo7J2DsF"}
[sync] [08-27|22:34:04.967] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/auth_batch.go:66 enqueued batch for processing during done
[sync] [08-27|22:34:04.967] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/syncervm_server.go:47 parsed state summary {"summary": "1168:2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT root=2gKBrSishaN73WRRybDV2TBU1Fkk1xK5DemCRs6GnUxo7J2DsF"}
[sync] [08-27|22:34:04.967] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/auth_batch.go:66 enqueued batch for processing during done
[sync] [08-27|22:34:04.967] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/syncervm_server.go:47 parsed state summary {"summary": "1168:2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT root=2gKBrSishaN73WRRybDV2TBU1Fkk1xK5DemCRs6GnUxo7J2DsF"}
[sync] [08-27|22:34:04.967] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/auth_batch.go:66 enqueued batch for processing during done
[sync] [08-27|22:34:04.968] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/syncervm_server.go:47 parsed state summary {"summary": "1168:2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT root=2gKBrSishaN73WRRybDV2TBU1Fkk1xK5DemCRs6GnUxo7J2DsF"}
[sync] [08-27|22:34:04.968] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/syncervm_client.go:111 starting state sync {"height": 1168, "summary": "1168:2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT root=2gKBrSishaN73WRRybDV2TBU1Fkk1xK5DemCRs6GnUxo7J2DsF", "already syncing": false}
[sync] [08-27|22:34:04.968] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/auth_batch.go:66 enqueued batch for processing during done
[sync] [08-27|22:34:04.968] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/resolutions.go:264 txs evicted from seen {"len": 0}
[sync] [08-27|22:34:04.968] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/vm.go:411 node is not ready yet
[sync] [08-27|22:34:04.968] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/resolutions.go:299 accepted block {"blkID": "2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT", "height": 1168, "txs": 4, "size": 852, "dropped mempool txs": 0, "state ready": false}
[sync] [08-27|22:34:04.968] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/resolutions.go:166 skipping unprocessed block {"height": 1168}
[sync] [08-27|22:34:04.968] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/resolutions.go:239 block processed {"blkID": "2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT", "height": 1168}
[sync] [08-27|22:34:07.195] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> vm/vm.go:634 parsed block {"id": "24QPD3VC4jdjHEhtqKrQJ5ipCfPuJMDkrP7KPzpxcJgkWkeJ8F", "height": 1169}
[sync] [08-27|22:34:07.196] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/block.go:769 block not processed {"height": 1168, "blkID": "2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT", "attemptVerify": true}
[sync] [08-27|22:34:07.196] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/block.go:798 block root does not match accepted state {"height": 1168, "blkID": "2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT", "accepted root": "2gKBrSishaN73WRRybDV2TBU1Fkk1xK5DemCRs6GnUxo7J2DsF", "block root": "pZ2QXErSL3RrT68HVfBk394S1oQfUtKAjMa97egFrHXePCx1F"}
[sync] [08-27|22:34:07.196] INFO <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/block.go:812 verifying block when view requested {"height": 1168, "blkID": "2WxsgcJfU19KGkucvq13YKqqSGfFzYo8Uot9MKoBsczf3tkuDT", "accepted": true}
[sync] [08-27|22:34:07.196] DEBUG <uJFYugQkWUt71MCMPpteLhho79QLEXd1kjG6vTdX9Z1YUh5JZ Chain> chain/block.go:428 could not get parent {"id": "2aJwYz2K6es3n3ZnVyKTXLquE6tdFACoZ3Cfw8pyRS13PzvDSL"
```

**Pull request merged**